pkgs/servers/readarr/default.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, stdenv, fetchurl, libmediainfo, sqlite, curl, makeWrapper, icu, dotnet-runtime, openssl, nixosTests }:
+
+let
+  os = if stdenv.isDarwin then "osx" else "linux";
+  arch = {
+    x86_64-linux = "x64";
+    aarch64-linux = "arm64";
+    x86_64-darwin = "x64";
+  }."${stdenv.hostPlatform.system}" or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+  hash = {
+    x64-linux_hash = "sha256-ufyHzCjqvbhz7EY/tsUgCFofGKQcsFp9wMkYqFAqXnE=";
+    arm64-linux_hash = "sha256-Z9K9XY+kJ7JEdBGJPqewodfjtQwQJZ+2LXBH8aQyb1c=";
+    x64-osx_hash = "sha256-zLwfAOXIjr3YCs7aVdC1spBhEkPld0ndBPueOTSgkN0=";
+  }."${arch}-${os}_hash";
+in stdenv.mkDerivation rec {
+  pname = "readarr";
+  version = "0.1.1.1320";
+
+  src = fetchurl {
+    url = "https://github.com/Readarr/Readarr/releases/download/v${version}/Readarr.develop.${version}.${os}-core-${arch}.tar.gz";
```

Comment:
Why not build the dotnet project from source?

Reply:
As before I followed was what done for the other *arr projects and they all downloaded the released archive.

I could try and see if we can succeed in building it from source. However I'm not that familiar with the dotnet environment. I would need some help in achieving this.

Reply:
I'm unaware of the background to building these from the archives. I remember that Jellyfin was also built like this at one point too, and in a7ccfa062cc it was updated to build from source. The current machinery for jellyfin might be a good example to copy for all the `*arr` servers in future, though I haven't tried it myself. If you felt like trying it for `readarr`, I could help switch the others over too. (I'm personally fine with the current approach in this PR in practice, so not trying to block it.)

Reply:
I have tried to build it from source yesterday, but I was not able to do so. I had to write my own `fetch-deps` script as the default one does not care about the `NuGet.config` of the repository and was theon unable to generate the correct deps.nix

Afterwards I was still hit by a lot of errors related to wildcards in version name not being compatible with `Determinism` as I don't know anything about dotnet I gave up there.

@SuperSandro2000 I propose that we go forward with installing from the archive and we could have a second PR trying torefactor all the *arr modules to be built from source and correct the various things that you pointed at in this PR. I would not mind helping @purcell on this he he has more experience than me on dotnet.